src/cholesky.jl

```diff
@@ -15,12 +15,12 @@ end
 
 Construct a `SkewCholesky` structure from the `UpperTriangular`
 matrix `Rm` and the permutation vector `Pv`. A matrix `Jm` of type `JMatrix`
-is build calling this function. 
+is build calling this function.
 The `SkewCholesky` structure has three arguments: `Rm`,`Jm` and `Pv`.
 """
 function SkewCholesky(Rm::UpperTriangular{<:T},Pv::AbstractVector{<:Integer}) where {T<:Real}
     n = size(Rm, 1)
-    return SkewCholesky{T,UpperTriangular{<:T},JMatrix{<:T},AbstractVector{<:Integer}}(Rm, JMatrix(T, n), Pv)
+    return SkewCholesky{T,typeof(Rm),JMatrix{T,+1},typeof(Pv)}(Rm, JMatrix{T,+1}(n), Pv)
```

Comment:
Note that `UpperTriangular{<:T}` etcetera are *not* concrete types.  You really want to use `typeof` here to return a `struct` with concretely typed fields.